### PR TITLE
MAT-9554 add feature flag AdminShareLibrary

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-9554](https://jira.cms.gov/browse/MAT-9554)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/hooks/useIsAdminShareLibraryEnabled.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { useFeatureFlags } from "./useFeatureFlags";
+import { useUserRoles } from "./useUserRoles";
+
+/**
+ * Custom hook to check if admin share library functionality is enabled for the current user.
+ * Returns true if the AdminShareLibrary feature flag is enabled AND the user has admin role.
+ */
+export function useIsAdminShareLibraryEnabled(): boolean {
```
Comment (adongare) [inline src/hooks/useIsAdminShareLibraryEnabled.ts]: with this pattern we have to add hooks for every admin permissions which is kind of overkill. moreover, util is supposed to be for common logic used in multiple apps and this is not used in multiple apps. so, I would rather do this in respective apps. what do you think?

Comment (sb-cecilialiu) [reply src/hooks/useIsAdminShareLibraryEnabled.ts]: @adongare how about change it with a parameter and do if..else..? There will be just one function this way.

Comment (sb-cecilialiu) [reply src/hooks/useIsAdminShareLibraryEnabled.ts]: @adongare I refactored the code and made a uniformed interface to handle all feature flags and user roles, can you please have a look again? Thanks!

**Review by adongare — APPROVED** [review]
